pkg/services/baremetal/remediation/remediation.go

```diff
@@ -53,6 +54,15 @@ func NewService(scope *scope.BareMetalRemediationScope) *Service {
 func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err error) {
 	host, err := s.getHost(ctx)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err = fmt.Errorf("HetzerBareMetalMachine not found")
+			if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
+				err = fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
+				record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
+				return res, err
```

Comment:
you don't want to return an error in this case I guess. You did an action (-> manipulated the machine object) and that's as expected.